src/main/daml/Daml/Finance/Lifecycle/ElectionEffect.daml

```diff
@@ -57,14 +56,13 @@ template ElectionEffect
         otherProduced
         settlementTime
 
-      calculate Effect.Calculate{actor; holdingCid} self = do
-        holding <- fetch holdingCid
-        getInstrument holding === targetInstrument
-        getAmount holding === amount
+      calculate Effect.Calculate{actor; quantity} self = do
+        quantity.unit === targetInstrument
+        assertMsg "Amount must match the elected amounts" $ amount == quantity.amount
         assertMsg "Election effects can only be calculated by the elector or their counterparty" $
           actor == owner || actor == custodian
         archive self
-        let scaleAll = fmap $ scale $ getAmount holding
+        let scaleAll = fmap $ scale amount
```

Comment:
In the case of an `Election Effect`, the purpose of `calculate` is to also verify that the holding amount does not exceed the elected amount.

This was done to prevent you from exercising 1000 units of an option and then claiming the effect with  2000 units.

Reply:
The amount check is still in there, it is just that the Claim rule now directly provides the quantity instead of passing it indirectly through the holdingCid. In the latter case, the ElectionEffect had to fetch the holdingCid as well to get the quantity, and thereby leaked some unnecessary information to the ElectionEffect provider. The purpose of this issue/PR was to avoid that.

Reply:
Maybe we could provide the `Calculate` as an interface method only, then the ElectionEffect provider would not even see what the quantity and who is calculating (the `Claim` rule would then simply invoke the calculate/scale method).